docs/naps/4-async-slicing.md

```diff
@@ -1,6 +1,6 @@
 (nap-4-async-slicing)=
 
-# NAP-4: Asynchronous slicing
+# NAP-4 — Asynchronous slicing
```

Comment:
This NAP currently is the only one with a colon, but is now joining the emdash family :p